### ISSUE TITLE
DHCP on non-admin NIC not working correctly during boot

### ISSUE THREAD

**Issue by sysconfig** [joyent/smartos-live#218]
In a system with more than one NIC, where the non-admin NIC is supposed to pick up its network configuration via DHCP during boot, the NIC fails to come up. Manual plumbing and (for example) "ifconfig e1000g0 dhcp start" works without problems, however.

This can be worked around by setting a network mask in /usbkey/config. However, setting a network mask while expecting to receive the network settings via DHCP defeats the purpose of DHCP.

This has been reproduced independently by myself and Robert Mustacchi (Joyent). See http://www.listbox.com/member/archive/184463/2013/05/sort/time_rev/page/1/entry/7:264/20130530200812:2C5A24EE-C986-11E2-8096-C09B003BCF0E/

In short, this works:

external_nic=aa:aa:aa:aa:aa:aa
external0_ip=dhcp
external0_netmask=255.255.255.0
[...]

This does not:

external_nic=aa:aa:aa:aa:aa:aa
external0_ip=dhcp
external0_netmask=
[...]

The interface tagged as "admin", however, does NOT require a pre-configured netmask with dhcp, as you would expect.

This can be reproduced on different servers and with admin/non-admin swapped, too.
